### ISSUE TITLE
Unexpected UI behavior for config vars

### ISSUE THREAD

**Issue by sdav9375** [hashicorp/waypoint#4039]
**Describe the bug**
When editing, deleting, and creating config vars in the UI, the results are unexpected. 

1) App-scoped vars cannot be edited, created or deleted in the UI, but there is no indication that this is impossible (edit and delete buttons are not disabled).
2) If you try to edit an app-scoped variable in the UI, it will instead create a new variable, with the given values, but at the project scope.

https://user-images.githubusercontent.com/17822084/195400116-a530cdfd-c324-4e91-b672-3643b2ca7bbf.mov

**Steps to Reproduce**
1) Try to edit an app-scoped variable in the UI and see that it doesn't edit it, but create a new one at the project scope.
2) Try to delete an app-scoped variable in the UI and see that it doesn't delete it and has no message that the delete failed.

**Expected behavior**
The UI should not have edit and delete options for app-scoped variables. Also, it would be nice to display the scope of the variable in the UI.

**Waypoint Platform Versions**
Additional version and platform information to help triage the issue if
applicable:

* Waypoint CLI Version: v0.10.2
* Waypoint Server Platform and Version: `docker`

**Comment by evanphx** [comment]
The issue here is related to how configVar ids are calculated. It seems like the Target on the ConfigVar isn't roundtripping from server to UI and back to server.

**Comment by voycey** [comment]
In the meantime is there a way to delete these via the CLI?

**Comment by briancain** [comment]
Yep @voycey https://tip.waypointproject.io/waypoint/docs/app-config/dynamic#unsetting-dynamic-values

It's not very obvious though! We've considering making an `unset` command.
